xmtp_mls/src/groups/mls_ext/mls_ext_welcome.rs

```diff
@@ -0,0 +1,67 @@
+use openmls::{
+    group::{MlsGroupJoinConfig, ProcessedWelcome, WireFormatPolicy},
+    prelude::{BasicCredential, MlsMessageBodyIn, MlsMessageIn, Welcome},
+};
+use tls_codec::Deserialize;
+
+use crate::{
+    client::ClientError, configuration::MAX_PAST_EPOCHS, groups::GroupError, hpke::decrypt_welcome,
+    identity::parse_credential, storage::xmtp_openmls_provider::XmtpOpenMlsProvider,
+};
+
+pub(crate) struct DecryptedWelcome {
+    pub(crate) welcome: Welcome,
+    pub(crate) added_by_inbox_id: String,
+}
+
+impl DecryptedWelcome {
+    pub(crate) fn from_encrypted_bytes(
+        provider: &XmtpOpenMlsProvider,
+        hpke_public_key: &[u8],
+        encrypted_welcome_bytes: &[u8],
+    ) -> Result<DecryptedWelcome, GroupError> {
+        tracing::info!("Trying to decrypt welcome");
+        let welcome_bytes = decrypt_welcome(provider, hpke_public_key, encrypted_welcome_bytes)?;
+
+        let welcome = deserialize_welcome(&welcome_bytes)?;
+
+        let join_config = build_group_join_config();
+
+        let processed_welcome =
+            ProcessedWelcome::new_from_welcome(provider, &join_config, welcome.clone())?;
+        let psks = processed_welcome.psks();
+        if !psks.is_empty() {
+            tracing::error!("No PSK support for welcome");
+            return Err(GroupError::NoPSKSupport);
+        }
+        let staged_welcome = processed_welcome.into_staged_welcome(provider, None)?;
+
+        let added_by_node = staged_welcome.welcome_sender()?;
+
+        let added_by_credential = BasicCredential::try_from(added_by_node.credential().clone())?;
+        let added_by_inbox_id = parse_credential(added_by_credential.identity())?;
+
+        Ok(DecryptedWelcome {
+            welcome,
```

Comment:
nit: 
i think we're being redundant here, we turn the welcome into a staged welcome on 37, then return the un-staged welcome. But where we use DecryptedWelcome, we immediately turn it back into a staged welcome

Reply:
Its a leftover from what the code was before, but would be nice to fix nonetheless